ssl/record/ssl3_record.c

```diff
@@ -286,21 +286,25 @@ int ssl3_get_record(SSL *s)
                 }
             } else {
                 /* SSLv3+ style record */
-                if (s->msg_callback)
```

Comment:
Is this callback only ever used for tracing?
If so moving it down is ok.

Reply:
Yep - it's only used for tracing.
I had to move it down so the "version" variable is populated, but added an extra call to msg_callback() for the case where decoding fails. It should behave the same as before.